### PR TITLE
RC 1 draft

### PR DESCRIPTION
**Pull request by scottrigby**
## Call to action

Please give your feedback here. Note we agreed to move to an RC process in order to make more rapid incremental progress toward a full release, while allowing for group feedback between RC releases.

Milestone dates reminder: tag [RC 1](https://github.com/open-gitops/documents/milestone/6) by this Friday, full [1.0.0](https://github.com/open-gitops/documents/milestone/1) release before KubeCon.

- Fixes: #13 
- Fixes: #18 
- Fixes: #20

Does not yet claim to fix this issue, but please keep this in mind during review. Accessibility is an ongoing process:
- #16 

## Context

⚠️ This RC 1 draft may feel a little like deja vu. It is closer to the initial principles as they have been all this time in the [WG README](https://github.com/gitops-working-group/gitops-working-group/blob/main/README.md#gitops-principles). 

Why? Over the past 6 months of meetings, we started by merging several of the initial principles. This was in part my doing, but it was an understandable direction at the time. As we began to update them from there with more precise wording as a group, the first few went smoothly. However, because each principle builds upon the previous ones, as we got closer to the end it became increasingly complex to figure out which pieces of which ideas should be in each, not quite being able to get all the way through them due to detailed debates each week. Important points were made by all. 

However the diminishing returns in later meetings is what prompted the decision to accelerate revising the principles through a RC/feedback process, where each RC attempts to integrate the very valuable points that were made by the group over the past months in a coherent way. 

In retrospect, a hypothesis is that merging several of the initial principles early on exacerbated the increasing complexity in bringing precise and consistent language to the later steps. This RC 1 draft attempts to solve that by rolling back to the initial version, and trying to make the fewest changes necessary to reconcile points and feedback from the principles committee that seem to have stood the test of time. So while this draft is very similar to the initial principles, there are some important changes that try to reconcile them with the points we made as a group. I hope you can see this in the draft by comparing the two (have a look at the initial commit for an easily viewable comparison of only those changes: f15f4ee3dda31745b12ed765886a843867d59d1b).

### Existing principle 5

We're not including this in the draft as of now because it's not clear how to explain it's necessity for GitOps. Why exactly was this principle here initially? Control theory language is valuable, but is it necessary here?

- Is it to say the software agents must take into account feedback/output from the observed actual state in deciding how or if to reconcile? If so, how is this different than what we already say in principle 4?
- Is it so say the software agents must take into account feedback/output from any previous reconciliation attempts (for example number of retries before stopping and notifying)?
- Since we are unclear on it's value add to the other 4 principles, we've left it out of the RC1 draft for now.
- Is this principle important for notifications?

5. **Operated in a closed loop**
    Software agents observe desired state and meta-state to take actions based on policy when the desired state cannot be reached. This may include things like notifications, rollbacks, etc.

### Potential new principle 6

We had discussed splitting this note from principle 4 into its own principle, to emphasize this as a critical point for GitOps. We are unclear whether we feel as a group this is already clear from the current wording of principle 4, or is this needs an additional explicit kick.

6. **The Source of Truth**
    Declared desired state will always take precedence over any other source of configuration.

## To-do

- [ ] Bring over co-authorship participation notes from #14 into a commit here, or when squashing this PR's commits
- [ ] Also add co-authorship credits to additional people who participate in this PR

### CODE REVIEW

File: GLOSSARY.md
Comment (RedbackThomson) [inline GLOSSARY.md:22]: Worth adding a note about the data in those systems not being a part of the desired state?

File: GLOSSARY.md
Comment (RedbackThomson) [inline GLOSSARY.md:30]: `s/sytem/system/g`

File: GLOSSARY.md
Comment (RedbackThomson) [inline GLOSSARY.md:37]: The second clause of this seems incomplete?

Comment (scottrigby) [reply GLOSSARY.md:37]: @RedbackThomson  I agree it reads incomplete as a paragraph, but I thought this could be addressed after RC 1 since it's been like this since February. See https://github.com/open-gitops/documents/pull/15#pullrequestreview-716397234

PS, while this glossary item was initially drafted with line breaks, I believe it was intended to be a list comprising the different parts of a software system as we currently understand it (see https://github.com/gitops-working-group/gitops-working-group/commit/ff540e1552e5a41c03a1fb317d554567777cdfe5). Like this:

> - **Software System:**
>     - One or more Runtime environments consisting of resources under management
>     - In each Runtime, management Agents to act on resources according to security policies
>     - One or more software Repositories for storing deployable artifacts that may be loaded into the runtime environments, eg. configuration files, code, binaries and packages
>     - One or more Administrators who are responsible for operating the runtime environments ie. installing, starting, stopping and updating software, code, configuration, etc
>     - A set of policies controlling access and management of repos, deployment, runtimes

Now that I say this, it would be an easy fix. Does this definition read more clearly as a list?

Comment (RedbackThomson) [reply GLOSSARY.md:37]: It does read more clearly as a list, although I was more specifically pointing out the following line:
> In each Runtime, management Agents to act on resources according to security policies

However, I understand if you want to leave the glossary as it stands for another time.

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md:22]: @RedbackThomson Ah that's right, this came up during the principles meetings. I had forgotten to account for that in this draft for RC 1. Do you want to [suggest](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request) an edit to this?

Comment (RedbackThomson) [reply GLOSSARY.md:22]: Perhaps

```suggestion
     The aggregate of all configuration data for a system form its _Desired State_ which is defined as data sufficient to recreate the system so that instances of the system are behaviourally indistinguishable, but does not include the state of any data stored within the system, eg. user data. 
```

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md:30]: @RedbackThomson would you please DM me your email for co-authorship purposes in the commit?

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md:37]: @RedbackThomson ok we can change to a list, since that's what it should have been all along. About that one list item, does this feel more clear/complete below?

```diff
- In each Runtime, management Agents to act on resources according to security policies
+ In each Runtime, the management Agents which act on resources according to security policies
```

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md:30]: For now I noted your username in the commit, and added a to-do item to the PR description to add co-author credits to this PR for everyone who participates. Thank you!

File: GLOSSARY.md
Comment (RedbackThomson) [reply GLOSSARY.md:37]: Yes :) That sounds good. Thanks

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md]: Got your email. Amended the commit to include your co-authored-by info 👍

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md]: How does this part look now?

Comment (RedbackThomson) [reply GLOSSARY.md]: Looks great, Scott. Thank you

**Review by RedbackThomson — APPROVED** [review]
Looks good to me, now!

File: PRINCIPLES.md
Comment (williamcaban) [inline PRINCIPLES.md:22]: The `attempt to apply` might not communicate the imperative need for the software agent to actually work towards the reconciliation. 

What about:
...observe actual system state and on [drift](./GLOSSARY.md#drift) [reconcile](./GLOSSARY.md#reconciliation) towards the desired state.

**Review by williamcaban — COMMENTED** [review]
comments inline

The rest looks good to me

File: PRINCIPLES.md
Comment (scottrigby) [reply PRINCIPLES.md:22]: I'd like to hear what you think about this section of the PR description above: "items left out of this PR"￼ > "existing principal 5"￼ > "5. Operated in a closed loop"? This is the principle that addresses a feedback loop based on configurations in the desired state and the output of attempts to apply to the actual state.￼

**Comment by moshloop** [comment]
LGTM

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:22]:
```suggestion
    The aggregate of all configuration data for a system form its desired state which is defined as data sufficient to recreate the system so that instances of the system are behaviourally indistinguishable, but do not include the state of any data stored within the system, eg. database contents. 
```

User data could mean lots of things so let's go with database contents. Also "do" not include since it's operating from data, which is plural.

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:30]:
```suggestion
    The process of ensuring that the actual state of a system matches its [desired state](#desired-state) declarations.
```
Not terribly needed given that versioned is covered.

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:39]:
```suggestion
  - In each runtime, the management agents which act on resources according to security policies
```

Fixing capitalization.

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:38]:
```suggestion
  - One or more runtime environments consisting of resources under management
```

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:40]:
```suggestion
  - One or more software repositories for storing deployable artifacts that may be loaded into the runtime environments, eg. configuration files, code, binaries, and packages
```

Comment (todaywasawesome) [reply GLOSSARY.md:40]: I will suggest removing this section and the one about administrators from the glossary in RC1. Not a show stopper.

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:49]:
```suggestion
- ## State Store

    A system for storing immutable versions of [Desired State](#desired-state) declarations.
    This state store should provide access control and auditing on the changes to the Desired State.
    Git, from which GitOps derives its name, is the canonical example used as this state store but any other system that meets these criteria may be used.
    In all cases, these state stores must be properly configured and special precautions must be taken to comply with requirements set out in the GitOps Principles.
```

Quick Grammar

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:7]:
```suggestion
    The temporary suspension of GitOps principles, often accomplished by pausing automated [reconciliation](#reconciliation).
```

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:14]:
```suggestion
    By "continuous" we adopt the industry standard to mean that [reconciliation](#reconciliation) continues to happen, not that it must be instantaneous.
```

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:26]:
```suggestion
    When a system's actual state changes for any reason other than its versioned [desired state](#desired-state) declarations having changed, we say that the system has drifted from its desired state.
```

File: GLOSSARY.md
Comment (todaywasawesome) [inline GLOSSARY.md:32]:
```suggestion
    Divergence could be due to the actual state unintentionally [drifting](#drift) from the desired state declarations, or a new desired state declaration version having been changed intentionally.
```

**Review by todaywasawesome — APPROVED** [review]
Removing "Break Glass" can wait until the RC feedback stage if you're more comfortable. I made some small grammar fixes and added hyperlinks for terms in the gloassary. I think we're good to move forward.

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md:30]: Agreed, "versioned" was never an important part of this glossary item, and is addressed explicitly in principle 2 👍

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md]: Agree with what you said here https://github.com/open-gitops/documents/pull/21#pullrequestreview-755646124:

> Removing "Break Glass" can wait until the RC feedback stage if you're more comfortable... I think we're good to move forward.

Making minor grammar, formatting and link fixes is ok even after others have reviewed this PR, but I am more comfortable leaving this as-is for now and discussing your point as part of the feedback for RC 1, given that the goal of that is to get wider feedback through an RC/feedback process from the group (including us) and wider community. Otherwise I think we'd need to wait for re-review by everyone on this PR. Instead let's meet the [milestone to release RC 1 today](https://github.com/open-gitops/documents/milestone/6).

File: PRINCIPLES.md
Comment (scottrigby) [reply PRINCIPLES.md:22]: Sounds like this is not a blocker for RC 1. @williamcaban let's continue to discuss this during the process when we send out RC 1 for feedback for folks. Most likely that will be in the form of a GitHub Discussion ❤️

File: GLOSSARY.md
Comment (scottrigby) [reply GLOSSARY.md]: OK, I'll "unresolve" this comment to remind you to bring that up during the process of discussing feedback on RC 1 🕐

**Review by csand-msft — APPROVED** [review]
Approve RC1

**Review by murillodigital — APPROVED** [review]
LGTM! 🎉🎉🎉

**Comment by csantanapr** [comment]
LGTM

**Comment by scottrigby** [comment]
Thanks everyone! Please see the squashed commit message for detailed attribution: https://github.com/open-gitops/documents/commit/11b362b799b41948e1b845aa90bf417ffe4dabfe

RC 1 tag forthcoming

**Comment by scottrigby** [comment]
Tag created! 🎉  https://github.com/open-gitops/documents/releases/tag/v1.0.0-rc.1

Please broadcast widely for feedback from the wider GitOps community 📣 🦄 ✨

File: PRINCIPLES.md
Comment (williamcaban) [reply PRINCIPLES.md:22]: My concern with `attempt to apply` is that it could open the interpretation that a series of remediations were attempted (lets say remediation A & B) and even if not successful it can `attempt` additional ones (eg. remediations C & D) which might lead to a situation of unrecoverable drift state (which brings us back to the snowflakes configurations results and desired state vs running state open to divergence)

How can we phrase it in a way that an `attempt` is not enough, but that it must continue trying remediate?

Comment (williamcaban) [reply PRINCIPLES.md:22]: WRT leaving principle 5, agree as it might be covered by principle 3. At least I have not found any extremely compelling example not covers by the other principles. The more concise the better.

**Review by lloydchang — COMMENTED** [review]
LGTM

**Pull request merged**
